### ISSUE TITLE
gometalinter: CloseNotifier deprecated

### ISSUE THREAD

**Issue by prusnak** [trezor/trezord-go#138]
```
server/api/api.go:71:15:warning: http.CloseNotifier is deprecated: the CloseNotifier interface predates Go's context package. New code should use Request.Context instead.  (SA1019) (staticcheck)
server/api/api.go:202:15:warning: http.CloseNotifier is deprecated: the CloseNotifier interface predates Go's context package. New code should use Request.Context instead.  (SA1019) (staticcheck)
```

**Comment by karel-3d** [comment]
Fixing this will maybe break old go versions that we still support. (but I
haven't looked that hard)

**Comment by prusnak** [comment]
I am fine with keeping this as is. Just created an issue, so we know why gometalinter test is failing.

**Comment by karel-3d** [comment]
My own q/a on SO :)

https://stackoverflow.com/questions/54890809/how-to-use-request-context-instead-of-closenotifier/54890870#54890870

**Comment by prusnak** [comment]
So is this change backwards compatible or in case it is not, how old versions will this break?

**Comment by karel-3d** [comment]
Looking at travis results from #146 and the go release notes, it seems it was introduced in 1.7; so all our tests are still working and #146 can be merged

**Comment by karel-3d** [comment]
Fixed by #146
